internal/registry/factory_test.go

```diff
@@ -860,3 +861,49 @@ func TestFactoryNewTargetLoaderWebConnectivity(t *testing.T) {
 		t.Fatal("expected zero length targets")
 	}
 }
+
+// customConfig is a custom config for [TestFactoryCustomTargetLoaderForRicherInput].
+type customConfig struct{}
+
+// customTargetLoader is a custom target loader for [TestFactoryCustomTargetLoaderForRicherInput].
+type customTargetLoader struct{}
+
+// Load implements [model.ExperimentTargetLoader].
+func (c *customTargetLoader) Load(ctx context.Context) ([]model.ExperimentTarget, error) {
+	panic("should not be called")
+}
+
+func TestFactoryCustomTargetLoaderForRicherInput(t *testing.T) {
+	// create factory creating a custom target loader
+	factory := &Factory{
+		build:            nil,
+		canonicalName:    "",
+		config:           &customConfig{},
+		enabledByDefault: false,
+		inputPolicy:      "",
+		interruptible:    false,
+		newLoader: func(config *targetloading.Loader, options any) model.ExperimentTargetLoader {
```

Comment:
I think we should have another test case similar to this one where we set `newLoader` as `nil` and assert that we get a valid loader from the passed in `ExperimentTargetLoaderConfig`

Reply:
✔️